pkg/log/zap/flags.go

```diff
@@ -60,26 +60,63 @@ func (ev *encoderFlag) Set(flagValue string) error {
 	val := strings.ToLower(flagValue)
 	switch val {
 	case "json":
-		ev.setFunc(newJSONEncoder())
+		ev.setFunc(newJSONEncoder)
 	case "console":
-		ev.setFunc(newConsoleEncoder())
+		ev.setFunc(newConsoleEncoder)
 	default:
 		return fmt.Errorf("invalid encoder value \"%s\"", flagValue)
 	}
 	ev.value = flagValue
 	return nil
 }
 
-func newJSONEncoder() zapcore.Encoder {
+func newJSONEncoder(ecfs []EncoderConfigOption) zapcore.Encoder {
 	encoderConfig := zap.NewProductionEncoderConfig()
+	for _, f := range ecfs {
+		f(&encoderConfig)
+	}
 	return zapcore.NewJSONEncoder(encoderConfig)
 }
 
-func newConsoleEncoder() zapcore.Encoder {
+func newConsoleEncoder(ecfs []EncoderConfigOption) zapcore.Encoder {
 	encoderConfig := zap.NewDevelopmentEncoderConfig()
+	for _, f := range ecfs {
+		f(&encoderConfig)
+	}
 	return zapcore.NewConsoleEncoder(encoderConfig)
 }
 
+type timeEncoderFlag struct {
```

Comment:
Unless I'm missing something, this flag seems to be doing something that could be accomplished by an `encoderFlag` since a `zapcore.TimeEncoder` is a field on `zapcore.EncoderConfig` right?

Reply:
As far as i understand, What you said is true for providing `zapcore.TImeEncoder` programmatically. `timeEncoderFlag` parses from `--zap-time-encoder=millis` flag directly, and applies to the `--zap-encoder=console` flag and then create the encoder.